src/QMCDrivers/QMCDriverNew.cpp

```diff
@@ -253,12 +254,12 @@ void QMCDriverNew::recordBlock(int block)
 
 bool QMCDriverNew::finalize(int block, bool dumpwalkers)
 {
-  population_.saveWalkerConfigurations();
-  MCPopulation::setWalkerOffsets(population_.getWalkerConfigsRef(), myComm);
+  population_.saveWalkerConfigurations(walker_configs_ref_);
```

Comment:
This could be a good time to use our version switch or a trait on the driver class to specialize for the batched drivers in QMCApp::runQMC putWalkers, process and other obsolete API methods come from there.

Reply:
I don't know how you will deal with WFOpt. It needs to write configurations back to the main state, crossing the WFOpt driver layer. Feel free to try it and see if we can get something better.